mlrun/serving/remote.py

```diff
@@ -205,8 +215,9 @@ def do_event(self, event):
 
     def _generate_request(self, event, body):
         method = self.method or event.method or "POST"
+        if self._headers_function_handler(body):
+            self.headers = self._headers_function_handler(body)
         headers = self.headers or {}
```

Comment:
`self.headers` shouldn't be mutated here.
```suggestion
            headers = self._headers_function_handler(body)
        else:
            headers = self.headers or {}
```